### ISSUE TITLE
DoubleQuoteUsage: braces inside double quotes not allowed

### ISSUE THREAD

**Issue by robbieaverill** [squizlabs/PHP_CodeSniffer#958]
I'm not sure if this a bug or a deliberate change, but I noticed this when upgrading from 2.4.0 to 2.6.0 recently (also evident in 2.5.1):

```php
<?php
$name = 'John';
echo "Hello {$name}, welcome back!";
```

Reports that string concatenation should be used instead. Is this a mistake, or a deliberate change? I'm always one to advocate single quotes wherever possible, but using braces in double quotes would be a valid use from my point of view.

**Comment by aik099** [comment]
It's valid from syntax point of view, but sniff doesn't like it. What you can't rewrite to avoid double quote usage is this:

```php
$a = "something doesn't work";
```

All other double quote usages are reported as error/warning.

**Comment by robbieaverill** [comment]
Fair enough. It doesn't conform to the Zend standard (which is fine,
because it's not part of it), and I couldn't see an equivalent sniff for
Zend- can I pull request a Zend alternative?

On Tuesday, 19 April 2016, Alexander Obuhovich <notifications@github.com>
wrote:

> It's valid from syntax point of view, but sniff doesn't like it. What you
> can't rewrite to avoid double quote usage is this:
>
> $a = "something doesn't work";
>
> All other double quote usages are reported as error/warning.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/squizlabs/PHP_CodeSniffer/issues/958#issuecomment-211770798>
>


-- 
*Robbie Averill*
LA PETITE MANOUCHE - NZ Gypsy Jazz Guitar Duo
Phone: +64 27 321 4758
contact@lapetitemanouche.com
www.lapetitemanouche.com

**Comment by aik099** [comment]
The `Zend` standard that comes with PHP_CodeSniffer doesn't match what current Zend standard is. You can send new sniff to it (via PR) if you'd like however.

**Comment by robbieaverill** [comment]
Good to know, thanks.

On Tuesday, 19 April 2016, Alexander Obuhovich <notifications@github.com>
wrote:

> The Zend standard that comes with PHP_CodeSniffer doesn't match what
> current Zend standard is. You can send new sniff to it (via PR) if you'd
> like however.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/squizlabs/PHP_CodeSniffer/issues/958#issuecomment-211776288>
>


-- 
*Robbie Averill*
LA PETITE MANOUCHE - NZ Gypsy Jazz Guitar Duo
Phone: +64 27 321 4758
contact@lapetitemanouche.com
www.lapetitemanouche.com
